ouroboros-consensus/src/Ouroboros/Consensus/HardFork/Combinator/Basics.hs

```diff
@@ -141,10 +146,29 @@ data HardForkLedgerConfig xs = HardForkLedgerConfig {
     }
   deriving (Generic)
 
+instance
+  ( SListI xs
+  , SerialiseNodeToClient (HardForkBlock xs) (PerEraLedgerConfig xs)
+  ) => SerialiseNodeToClient (HardForkBlock xs) (HardForkLedgerConfig xs) where
+  encodeNodeToClient ccfg version (HardForkLedgerConfig hflcShape perEraLedgerConfig)
+    = mconcat [
+            encodeListLen 2
+          , encodeNodeToClient @_ @(History.Shape xs)      ccfg version hflcShape
+          , encodeNodeToClient @_ @(PerEraLedgerConfig xs) ccfg version perEraLedgerConfig
+          ]
+  decodeNodeToClient ccfg version = do
+      enforceSize "HardForkLedgerConfig" 2
+      HardForkLedgerConfig
+        <$> decodeNodeToClient @_ @(History.Shape xs)       ccfg version
+        <*> decodeNodeToClient @_ @(PerEraLedgerConfig xs)  ccfg version
+
 instance CanHardFork xs => NoThunks (HardForkLedgerConfig xs)
 
 type instance LedgerCfg (LedgerState (HardForkBlock xs)) = HardForkLedgerConfig xs
 
+instance (CanHardFork xs, UpdateLedger (HardForkBlock xs))
+  => HasPartialLedgerConfig (HardForkBlock xs) where
```

Comment:
Well, I made terrible progress here in the 24 days since I wrote that.

So I'll just ask you to help me: why is this instance now necessary? I generally think of `HasPartialLedgerConfig` as something an era's block type needs before you can put that era into the HFC. And we generally do not plan to nest HFC. So this instance smells wrong to me.

Why do we need it now?

Reply:
Ooops you're actually correct in that I can remove this. I think it is just a vestigial change from a previous revision where I was sending `PartialLedgerConfig` and hence required that all block types have a `HasPartialLedgerConfig` instance. Now we're sending `LedgerConfig` directly and no longer require this instance.